typing/printtyp.ml

```diff
@@ -1139,18 +1147,14 @@ let rec tree_of_typexp mode ty =
             if closed && all_present then
               out_variant
             else
-              let non_gen = is_non_gen mode (Transient_expr.type_expr px) in
               let tags =
                 if all_present then None else Some (List.map fst present) in
-              Otyp_variant (non_gen, Ovar_typ out_variant, closed, tags)
+              Otyp_variant (Ovar_typ out_variant, closed, tags)
         | _ ->
-            let non_gen =
-              not (closed && all_present) &&
-              is_non_gen mode (Transient_expr.type_expr px) in
```

Comment:
The condition is currently redundant: 

- `ty` is a `Tvariant _` here
- `px` is `proxy ty` and `proxy ty` is either:
  - the row variable if it is not a "static row", and a "static row" means `closed && all_present`.
  - the type `ty` itself otherwise
- `is_non_gen mode ty` tests for non generalizable type variables, in particular it imples `isTvar ty`.

Consequently, `is_non_gen mode (Transient_expr.type_expr px)` implies `not (closed && all_present)`.